natlib/svcd.c

```diff
@@ -30,6 +31,52 @@ static const LUA_REG_TYPE svcd_meta_map[] =
 // Maintainer: Michael Andersen <michael@steelcode.com>
 /////////////////////////////////////////////////////////////
 
+//Lua: storm.n.svcd_advert_received(pay, srcip, srcport)
+//Dispatch an advertisment
+
+int svcd_advert_received(lua_State *L) 
+{
+	char *pay = luaL_checkstring(L, 1);
+	char *srcip = luaL_checkstring(L, 2);
+	int srcport = luaL_checknumber(L, 3);
+        int pack_result = libmsgpack_mp_unpack(L);
```

Comment:
You should use `lua_pushlightfunction(L, libmsgpack_mp_unpack)` instead of calling the c function directly.  Use lua to handle the stack for you (remember Gabe said to not call C functions)

men in #000000